src/main/groovy/com/github/winplay02/gitcraft/manifest/skyrising/SkyrisingMetadataProvider.java

```diff
@@ -0,0 +1,119 @@
+package com.github.winplay02.gitcraft.manifest.skyrising;
+
+import com.github.winplay02.gitcraft.manifest.BaseMetadataProvider;
+import com.github.winplay02.gitcraft.manifest.ManifestSource;
+import com.github.winplay02.gitcraft.meta.VersionDetails;
+import com.github.winplay02.gitcraft.meta.VersionInfo;
+import com.github.winplay02.gitcraft.types.OrderedVersion;
+import com.github.winplay02.gitcraft.util.MiscHelper;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+
+public class SkyrisingMetadataProvider extends BaseMetadataProvider<SkyrisingManifest, SkyrisingManifest.VersionEntry> {
+
+	private final Map<String, VersionDetails> versionDetails = new HashMap<>();
+
+	public SkyrisingMetadataProvider() {
+		this.addManifestSource("https://skyrising.github.io/mc-versions/version_manifest.json", SkyrisingManifest.class);
+	}
+
+	@Override
+	public ManifestSource getSource() {
+		return ManifestSource.SKYRISING;
+	}
+
+	@Override
+	public String getName() {
+		return "Skyrising Version Metadata (https://skyrising.github.io/mc-versions/)";
+	}
+
+	@Override
+	public String getInternalName() {
+		return "skyrising";
+	}
+
+	@Override
+	protected OrderedVersion loadVersionFromManifest(SkyrisingManifest.VersionEntry manifestEntry, Path targetDir) throws IOException {
+		VersionInfo info = this.fetchVersionMetadata(manifestEntry.id(), manifestEntry.url(), manifestEntry.sha1(), targetDir.resolve("info"), "version info", VersionInfo.class);
+		VersionDetails details = this.fetchVersionMetadata(manifestEntry.id(), manifestEntry.details(), null, targetDir.resolve("details"), "version details", VersionDetails.class);
```

Comment:
I was debugging a problem here for way too long, and it turned out to just be stale version details. Since https://github.com/skyrising/mc-versions/issues/4 hasn't been getting any response, checking the hash wouldn't be possible right now. Is there another way to check if the details are still up-to-date? My first attempt was using the GitHub API to fetch the blobs from the repository (including a hash) but I quickly hit the rate limit, so this isn't feasible.

Reply:
Hmm nothing comes to mind. You can do some basic checks of course like checking that the file is not empty and parses to valid JSON, but that doesn't verify in any way that the data itself is not stale, in case the file was modified in the source.